include/onnxruntime/core/framework/kernel_def_builder.h

```diff
@@ -56,17 +61,20 @@ class KernelDef {
     return alias_map_;
   }
 
-  const MemTypeMap& InputMemoryType() const {
-    return input_memory_type_args_;
+  OrtMemType InputMemoryType(size_t input_index) const {
+    auto it = input_memory_type_args_.find(input_index);
+    if (it == input_memory_type_args_.end())
+      return default_inputs_mem_type_;
+    else
+      return it->second;
   }
 
-  const MemTypeMap& OutputMemoryType() const {
-    return output_memory_type_args_;
-  }
-
-  // legacy interface for winml, should not be used in onnxruntime
-  const MemTypeMap& MemoryType() const {
-    return output_memory_type_args_;
+  OrtMemType OutputMemoryType(size_t output_index) const {
+    auto it = output_memory_type_args_.find(output_index);
```

Comment:
this is not introduced by your PR, but why we use map to store the types, instead of vector.

Reply:
I guess in most case people only declare memory type for very few input/output, map seems more suitable.